### DISCUSSION TITLE
Kubernetes setup of keygen self host

### DISCUSSION THREAD

**Discussion by polasomeshwar** [keygen-sh/community#138] [Q&A]
Hi everyone , I want to setup keygen self host as kubernetes deployments but not sure how to proceed, since we are dealing with environment variables , please help me with this.

**Comment by ezekg** [comment]
We don't use k8s, so this is a bit out of my wheel house. You may have some luck asking in [our Discord](https://keygen.sh/discord/), but other than that, I can ask around and see if any of our Keygen EE customers use k8s in their setup. Is there anything you're particularly stuck on? Why doesn't k8s play well with environment variables? It [should work fine with a twelve-factor app](https://www.redhat.com/architect/12-factor-app-containers) like Keygen.
